### PR TITLE
What's New in DevTools 126

### PR DESCRIPTION
**Pull request by leahmsft**
Rendered article for review: 
* **What's New**
   * https://review.learn.microsoft.com/microsoft-edge/devtools-guide-chromium/whats-new/2024/06/devtools-126?branch=pr-en-us-3192
   * [Planned live](https://learn.microsoft.com/microsoft-edge/devtools-guide-chromium/whats-new/2024/06/devtools-126)
* **Experiments**
   * https://review.learn.microsoft.com/microsoft-edge/devtools-guide-chromium/experimental-features/?branch=pr-en-us-3192
   * [Before/Live](https://learn.microsoft.com/microsoft-edge/devtools-guide-chromium/experimental-features/)

AB#51753911

### CODE REVIEW

File: microsoft-edge/devtools-guide-chromium/whats-new/2024/06/devtools-126.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+<!-- Subtitle: The DevTools extensions that you install will display their icons so they're easily identifiable in the Activity Bar.-->
+
+In previous versions of Microsoft Edge, there was a bug where some browser extensions were missing an icon when opened in DevTools.  For example, when the React extension was activated, the icon for the extension in the **Activity Bar** was displayed as a grey box.
+
+<!-- todo: add screenshot if can get one from Stable 124 -->
+
+This has been fixed; icons for browser extensions are correctly displayed in the **Activity Bar**.
+
+![Display React extension icons](./devtools-126-images/react-extension-icon.png)
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/whats-new/2024/06/devtools-126.md:27]: is it correct that there are two icons for React extension: one big, then one small?
@leahmsft 

rendered article:
https://review.learn.microsoft.com/microsoft-edge/devtools-guide-chromium/whats-new/2024/06/devtools-126?branch=pr-en-us-3192

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 2]
```diff
@@ -649,13 +667,39 @@ See also:
...
+
+
+<!-- ====================================================================== -->
+## Enable new Performance panel landing page which includes live metrics and field data
+<!-- checkbox has no (?) link pointing to this anchor wording -->
+
+_new as of Microsoft Edge 128_
+
+In the **Performance** tool, displays a main page section that shows live metrics and field data.
+<!-- todo: need real desc -->
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:690]: Enable new Performance panel landing page which includes live metrics and field data
Need real desc @captainbrosset

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 3]
```diff
@@ -811,77 +855,77 @@ See also:
...
 
 
 <!-- ====================================================================== -->
-## Timeline: invalidation tracking
-<!-- checkbox has a (?) link to this anchor wording -->
+## Performance panel: invalidation tracking
+<!-- checkbox has no (?) link pointing to this anchor wording -->
 
 Controls whether to show invalidation tracking in the **Performance** tool.
+<!-- todo: real desc for this older entry -->
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:867]: Performance panel: invalidation tracking
Need real desc @captainbrosset

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 4]
```diff
@@ -904,30 +948,98 @@ See also:
...
+
+
+<!-- ====================================================================== -->
+## Performance panel: Enable collecting enhanced traces
+<!-- checkbox has no (?) link pointing to this anchor wording -->
+
+_new as of Microsoft Edge 128_
+
+In the **Performance** tool, when recording, collects enhanced traces.
+<!-- todo: real desc - maybe extract from article -->
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:988]: Performance panel: Enable collecting enhanced traces
Need real desc @captainbrosset
try from article:
https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/experimental-features/share-traces?branch=pr-en-us-3192

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 5]
```diff
@@ -904,30 +948,98 @@ See also:
...
 
+
+<!-- ====================================================================== -->
+## Performance panel: Enable collecting source text for compiled script
+<!-- checkbox has no (?) link pointing to this anchor wording -->
+
+_new as of Microsoft Edge 128_
+
+In the **Performance** tool, when recording, collects source text (the original source code) for compiled JavaScript code.
+<!-- todo: real desc -->
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1007]: Performance panel: Enable collecting source text for compiled script
Need real desc @captainbrosset
check https://review.learn.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/experimental-features/share-traces?branch=pr-en-us-3192

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 6]
```diff
@@ -904,30 +948,98 @@ See also:
...
+
+Status:
+* This checkbox is present in Microsoft Edge Canary 128.
+* This checkbox is not present in Microsoft Edge Stable 125.
+
+See also:
+* [Share enhanced performance and memory traces](../experimental-features/share-traces.md)
+* [Performance features reference](../evaluate-performance/reference.md)
+* [Performance recording event reference](../evaluate-performance/performance-reference.md)
+<!-- todo: best articles? -->
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1017]: best See Also's?
@captainbrosset

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 7]
```diff
@@ -904,30 +948,98 @@ See also:
...
-* This checkbox is present in Microsoft Edge Canary 125.
-* This checkbox is not present in Microsoft Edge Stable 124.
+* This checkbox is present in Microsoft Edge Canary 128.
+* This checkbox is not present in Microsoft Edge Stable 125.
+
+See also:
+* [Share enhanced performance and memory traces](../experimental-features/share-traces.md)
+* [Performance features reference](../evaluate-performance/reference.md)
+* [Performance recording event reference](../evaluate-performance/performance-reference.md)
+<!-- todo: best articles? -->
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1037]: best See Also's for Debug Mode?
@captainbrosset

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 8]
```diff
@@ -1003,21 +1124,24 @@ See also:
...
+## Enable the experimental, WIP sidebar for the Performance Panel.
+<!-- checkbox has no (?) link pointing to this anchor wording -->
+<!-- checkbox label has period -->
 
-_new as of Microsoft Edge 126_
+_new as of Microsoft Edge 128_
 
-An internal setting that controls whether to enable the legacy tracing model that the **Performance** tool uses.  Do not use.
+In the **Performance** tool, enables the Windows Information Protection (WIP) sidebar.
+<!-- todo: real desc - check article -->
```
Comment (mikehoffms) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1139]: Enable the experimental, WIP sidebar for the Performance Panel.
need real desc @captainbrosset - expand "WIP" correctly.
check https://learn.microsoft.com/en-us/windows/security/information-protection/windows-information-protection/protect-enterprise-data-using-wip

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 3 above)
Comment (captainbrosset) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:867]: Adds an **Invalidations** section for **Recalculate Style** events under the **Summary** tab of the **Performance** tool. The **Invalidations** section provides information about the root cause for a **Recalculate Style** event, which might be helpful when investigating long **Recalculate Style** events that are causing performance issues.

The **Invalidations** section lists the DOM nodes that got invalidated as a result of a change on the page, such as a DOM mutation, and which the browser engine had to restyle during the **Recalculate Style** event.

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 4 above)
Comment (captainbrosset) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:988]: @robpaveza can you help with this one please? Isn't this something that Edge has been able to do for some time, and we're upstreaming it now. Trying to understand whether enabling this experiment downstream does anything, or whether we just always support enhanced traces.

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 5 above)
Comment (captainbrosset) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1007]: My limited understanding of this experiment is that it doesn't have an impact on the UI of DevTools. It's used to make it possible to store the source code of compiled scripts in the trace file, when exporting it.

@robpaveza: can you please help with this one too? Is this used when loading an enhanced trace in a new window, perhaps?

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 6 above)
Comment (captainbrosset) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1017]: I think the article about sharing enhanced traces is the best See Also we can add here.

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 8 above)
Comment (captainbrosset) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1139]: This experiment is a work in progress (WIP). Enabling it adds a vertical sidebar to the **Performance** tool. The sidebar will later contain performance insights helping you find bottlenecks in your recorded performance traces more quickly.

Comment (captainbrosset) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1139]: Let's not add any see also here, I can't think of any that would work.

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 7 above)
Comment (captainbrosset) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1037]: Adds a **Trace Event** section to the **Summary** tab of the **Performance** tool. The **Trace Event** section is displayed when you select an event in a recorded performance trace. The **Trace Event** section displays debugging information about the selected trace event, such as its name and duration.

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 8 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1139]:
```
Enables the sidebar in the **Performance** tool.  The sidebar is an experimental work in progress (WIP).

<!-- omit See Also -->

<!-- todo: screenshot of resulting UI, eg in Canary -->
```
@captainbrosset fyi i'm not seeing the resulting sidebar in the ui, so no screenshot

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 5 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1007]:
```
In the **Performance** tool, when exporting the trace file, stores the source code of 
compiled scripts in the trace file.  <!-- This experiment is used when loading an enhanced 
trace in a new window. -->
```

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 3 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:867]:
```
Adds an **Invalidations** section for **Recalculate Style** events in the **Summary** tab 
of the **Performance** tool.  The **Invalidations** section provides information about 
the root cause for a **Recalculate Style** event, which might be helpful when 
investigating long-running **Recalculate Style** events that are causing performance 
issues.

The **Invalidations** section lists the DOM nodes that were invalidated as a result of 
a change on the webpage, such as a DOM mutation, and which the browser engine 
had to restyle during the **Recalculate Style** event.
```

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 2 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:690]:
```
In the **Performance** tool, adds a landing page that's displayed when no performance traces 
have been recorded yet.  The landing page displays performance metrics about the rendered 
page that are updated live, such as:
* **Largest Contentful Paint (LCP)**
* **Cumulative Layout Shift (CLS)**
* **Interaction to Next Paint (INP)**
```

**Review by mikehoffms — APPROVED** [review]

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 5 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md]: Jayson to assist

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 4 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1000]: Jayson to assist

Comment (ja-why-son) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1000]: This experiment is for performance panel capturing newly added traces to support enhanced traces v2 (currently implementing in upstream). It does not affect the ability of downstream enhanced traces and the state of it doesn't impact downstream user's experience.

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 5 above)
Comment (ja-why-son) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md]: Same as the other one, this experiment is for performance panel capturing newly added traces for script source text to support enhanced traces v2 (currently implementing in upstream). It does not affect the ability of downstream enhanced traces and the state of it doesn't impact downstream user's experience.

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 4 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1000]: result:
```
This experiment makes the **Performance** tool capture newly added traces, to support 
enhanced traces v2.  This feature is being implemented in upstream.  This experiment 
doesn't affect the capabilities of downstream enhanced traces.  The state of this checkbox 
doesn't impact a downstream user's experience.
```

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md (hunk 5 above)
Comment (mikehoffms) [reply microsoft-edge/devtools-guide-chromium/experimental-features/index.md]: result:
```
In the **Performance** tool, when exporting the trace file, stores the source code of 
compiled scripts in the trace file.

This experiment makes the **Performance** tool capture newly added traces for script 
source text, to support enhanced traces v2.  This feature is being implemented in 
upstream.  This experiment doesn't affect the capabilities of downstream enhanced 
traces.  The state of this checkbox doesn't impact a downstream user's experience.
```

**Review by ja-why-son — APPROVED** [review]

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 9]
```diff
@@ -904,30 +960,95 @@ See also:
...
+See the above section, showing the newer label for the checkbox.
+
+
+<!-- ====================================================================== -->
+## Performance panel: Enable collecting enhanced traces
+<!-- checkbox has no (?) link pointing to this anchor wording -->
+
+_new as of Microsoft Edge 128_
+
+This experiment makes the **Performance** tool capture newly added traces, to support enhanced traces v2.  This feature is being implemented in upstream.  This experiment doesn't affect the capabilities of downstream enhanced traces.  The state of this checkbox doesn't impact a downstream user's experience.
```
Comment (captainbrosset) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:999]:
```suggestion
This experiment makes the **Performance** tool capture newly added traces, to support enhanced traces v2.  This feature is being implemented in the upstream Chromium project.  This experiment doesn't affect the capabilities of Microsoft Edge to capture enhanced traces and doesn't impact DevTools user's experience.
```

File: microsoft-edge/devtools-guide-chromium/experimental-features/index.md [hunk 10]
```diff
@@ -904,30 +960,95 @@ See also:
...
+
+<!-- ====================================================================== -->
+## Performance panel: Enable collecting source text for compiled script
+<!-- checkbox has no (?) link pointing to this anchor wording -->
+
+_new as of Microsoft Edge 128_
+
+In the **Performance** tool, when exporting the trace file, stores the source code of compiled scripts in the trace file.
+
+This experiment makes the **Performance** tool capture newly added traces for script source text, to support enhanced traces v2.  This feature is being implemented in upstream.  This experiment doesn't affect the capabilities of downstream enhanced traces.  The state of this checkbox doesn't impact a downstream user's experience.
```
Comment (captainbrosset) [inline microsoft-edge/devtools-guide-chromium/experimental-features/index.md:1019]:
```suggestion
This experiment makes the **Performance** tool record script source text in treaces.  This feature is being implemented in the upstream Chromium project.
```

**Review by captainbrosset — CHANGES_REQUESTED** [review]

**Review by captainbrosset — APPROVED** [review]

**Pull request merged**